lib/action_view/component/render_monkey_patch.rb

```diff
@@ -15,15 +15,27 @@ def render(options = {}, args = {}, &block)
 
           options.render_in(self, &block)
         elsif options.is_a?(Class) && options < ActionView::Component::Base
-          options.new(args).render_in(self, &block)
+          safe_new_component(options, args).render_in(self, &block)
         elsif options.is_a?(Hash) && options.has_key?(:component)
-          options[:component].new(options[:locals]).render_in(self, &block)
+          safe_new_component(options[:component], options[:locals]).render_in(self, &block)
         elsif options.respond_to?(:to_component_class) && !options.to_component_class.nil?
           options.to_component_class.new(options).render_in(self, &block)
         else
           super
         end
       end
+
+      private
+
+      def safe_new_component(component_class, options)
+        # Check the arity of `#initialize` to avoid passing an empty
+        # `options` hash when no arguments are expected. Also, check
+        # `options` to avoid dropping arguments passed to `#render`,
+        # which might hide a helpful `ArgumentError`.
+
+        arity = component_class.instance_method(:initialize).arity
+        (options.blank? && arity.zero?) ? component_class.new : component_class.new(options)
```

Comment:
It's a nitpick, but I'd prefer we avoid using a ternary here. Mind splitting this into `if else`?

Reply:
no prob!

Which do you prefer?

```
# option A

if options.blank? && arity.zero?
  component_class.new
else 
  component_class.new(options)
end
```

```
# option B

if options.blank? && arity.zero?
  return component_class.new
end

component_class.new(options)
```

```
# option C
return component_class.new if options.blank? && arity.zero?

component_class.new(options)
```

Reply:
🙇 A, please ❤️

Reply:
🤙